### ISSUE TITLE
Support for mediawiki?

### ISSUE THREAD

**Issue by ehmkah** [docToolchain/docToolchain#510]
**Is your feature request related to a problem? Please describe.**
Export to confluence is provided and export to mediawiki would be nice too have.

**Describe the solution you'd like**
A task `publishToMediaWiki` (or smiliar)  which publishes documentation to a media wiki. Configuration should be possible in Config.groovy like confluence.

**Describe alternatives you've considered**
Though of building something with panddoc outside of doctoolchain, but inside of doctoolchain it might be more useful.

**Additional context**
no. hope the description is good enough.

**Comment by rdmueller** [comment]
Interesting idea. 
So this, we would need some kind of converter from AsciiDoc to X. 
AFAIK, MediaWiki uses its own markup, does it?

**Comment by ehmkah** [comment]
Yes mediawiki has its own markup. https://www.mediawiki.org/wiki/Help:Formatting/en 

Pandoc supports mediawiki as output-format. (https://www.mediawiki.org/wiki/Help:Formatting/en )  maybe thats somehow useful

**Comment by rdmueller** [comment]
pandoc could be one good approach. What is the interface to create new pages in mediawiki? Is there an API or is just the filesystem the interface?

**Comment by ehmkah** [comment]
The mediawiki has an API (https://www.mediawiki.org/wiki/API:Main_page) which seems to be useable for creating pages. Pandoc seems to export mediawiki-files, so they need to be parsed and uploaded somehow. 

Do you think its difficult to integrate pandoc into the doctoolchain-flow? Its written in haskell.

**Comment by rdmueller** [comment]
pandoc is already used for docx and epub output but it is expected that it is already installed in the machine.

**Comment by rdmueller** [comment]
So, docToolchain would have to convert AsciiDoc to docBook (pandoc can't read AsciiDoc) and then, pandoc would convert this to mediaWiki which will be pushed through the API.

**Comment by rdmueller** [comment]
what is the best way to get a media wiki up and running quickly?

**Comment by ehmkah** [comment]
You are right pandoc does not support asciidoctor as input (i just saw it on the output). 

"Best" way to setup a media wiki, maybe with mediawikis provided docker-image as described in https://www.mediawiki.org/wiki/Docker/Hub and this is the image https://hub.docker.com/_/mediawiki Warning: I never did it this way, but I hope the documentation works

**Comment by rdmueller** [comment]
> You are right pandoc does not support asciidoctor as input

btw: this is not a problem since the docBook output contains afaik all information which is contained in AsciiDoc. These formats seem to be fully interchangable.

**Comment by rdmueller** [comment]
@ehmkah - can you please update this issue with your findings?

**Comment by ehmkah** [comment]
I tried to publish asciidoctor pages to mediawiki. The goal was to have something similar to "publishConfluence" task.  The approach was run "generateDocbook" then transform this docbook to mediawiki  with pandoc (i did it manually because I was in exploring mode) then create subpages for each heading1/2 then upload those subpages to mediawiki with help of (https://github.com/MER-C/wiki-java) .

it worked but I had 

- some problems with the links. Mediawiki has its own definition of anchors for headlins and therefore plain asciidoctor links do sometimes not work if you want to link to a specific headline
- not tried out to upload images
- problems with mediawikis API thats why I needed an other library

But anyway. Exporting the complete document to one mediawikipage is simplier and can be the first step to publish pages to mediawiki and leave space for future improvements.
